src/main/scala/ylabs/messaging/Client.scala

```diff
@@ -62,6 +62,16 @@ class Client extends FSM[State, Context] {
   lazy val domain = config.getString(ConfigKeys.domain)
   lazy val host = config.getString(ConfigKeys.host)
 
+  def splitUserIntoNameAndDomain(user: User):(String, String) = {
```

Comment:
to be consistent with the rest of the codebase you could return `(User, Domain)`, i.e. create a `case class Domain(value: String) extends AnyVal`